### PR TITLE
Secondary infections

### LINKED ISSUE #262: Keeping track of secondary infections
**Is your feature request related to a problem? Please describe.**
We wish to record the number of people each person infects over the course of the epidemic (each time they are re-infected) so that we can find the R_t value at each time step. The dataframes produced will be further used in the sampling package EpiOS.

**Describe the solution you'd like**
I will change the `Person` class to record a list of secondary infections each person has during the epidemic. I will also update the simulation to record and output a dataframe containing this information.

### PR DESCRIPTION
**Pull request by mghosh00**
## Summary

We wish to record secondary infections in pyEpiabm in order to calculate R_t, the effective reproduction rate. This has now been implemented with results being recorded to secondary_infections.csv. A template for this file can be found within the linked issue.

## Checklist

- [x] All new functions have docstrings in the correct style
- [ ] I've verified the complete docs build locally without errors
- [x] I've maintained 100% coverage (please mention any 'no cover' annotations explicitly)
- [x] I've unit-tested all new methods directly

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] I have updated the wiki with new parameters/model functionality

## Closing issues

Closes #262

### CODE REVIEW

**Comment by mghosh00** [comment]
r_t
simulation_flow_SIR_plot

**Review by KCGallagher — APPROVED** [review]
This is great - I can't spot anything I want changing! Only thing is, it might be nice to add to the python examples for infection history to collect & plot this data for a simple simulation (both the Rt over time, and distribution of secondary infections per person)

File: pyEpiabm/pyEpiabm/core/person.py [hunk 1]
```diff
@@ -278,3 +278,15 @@ def increment_num_times_infected(self):
...
+        """Increments the number of secondary infections the given person has
+        for this specific infection period (i.e. if the given person has been
+        infected multiple times, then we only increment the current secondary
+        infection count)
+        """
+        if self.secondary_infections_counts:
+            self.secondary_infections_counts[-1] += 1
+        else:
+            raise RuntimeError("Cannot call increment_secondary_infections "
+                               "while secondary_infections_counts is empty")
```
[the remark this answers is not in the document]
Comment (rccreswell) [reply pyEpiabm/pyEpiabm/core/person.py:292]: Does this do the same thing? It's not important at all, but all else equal python prefers try/except to if/else, as it is usually faster.

**Review by rccreswell — APPROVED** [review]

File: pyEpiabm/pyEpiabm/core/person.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mghosh00) [reply pyEpiabm/pyEpiabm/core/person.py:292]: Of course, will accept it!

**Pull request merged**